lightning/src/chain/keysinterface.rs

```diff
@@ -848,6 +858,12 @@ impl KeysManager {
 	/// Note that until the 0.1 release there is no guarantee of backward compatibility between
 	/// versions. Once the library is more fully supported, the docs will be updated to include a
 	/// detailed description of the guarantee.
+	///
+	/// This struct cannot be used for nodes that wish to support receiving multi-node or phantom
+	/// payments; [`PhantomKeysManager`] must be used instead.
+	///
+	/// Switching between this struct and [`PhantomKeysManager`] will invalidate any previously issued
+	/// invoices and attempts to pay previous invoices will fail.
```

Comment:
> I mean I think in general we don't want to support changing the return value of `get_inbound_payment_key_material` at all,

Yeah, that is what I meant by it being a more general concern.

> so I'm not sure we want to describe this in the public interface, but it is something for users of our own structs to be aware of, because we support "upgrading".

How is upgrading supported? Isn't this comment saying upgrading is _not_ supported?

Reply:
Though I suppose you could stop issuing invoices and wait for all the outstanding invoices' expiries to pass.

Reply:
Understood. By my original comment I meant we should say in `KeysInterface::get_inbound_payment_key_material` that changing an implementation to calculate this in a different way would lead to undefined behavior. Currently, it doesn't read as strict to me. Seems rather to indicate only that successive calls in a given execution should always return the same value.

Reply:
Ah, yes, sorry, I somehow understood your comment to say that we should describe the outcome of changing the return value in `KeysInterface::get_inbound_payment_key_material`, which I dont think we should do. stating it cannot change/if it does its undefined we should do, agreed.